message.go

```diff
@@ -485,27 +483,8 @@ func (ssa SingleSegmentArena) String() string {
 // Calling Release is optional; if not done the garbage collector
 // will release the memory per usual.
 func (ssa *SingleSegmentArena) Release() {
-	// Clear the memory, so there's no junk in here for the next use:
-	for i := range *ssa {
-		(*ssa)[i] = 0
-	}
-
-	// Truncate the segment, since we use the length as the marker for
-	// what's allocated:
-	(*ssa) = (*ssa)[:0]
-
-	singleSegmentPool.Put(ssa)
-}
-
-// Like SingleSegment, but doesn't use the pool
-func singleSegment(b []byte) *SingleSegmentArena {
-	return (*SingleSegmentArena)(&b)
-}
-
-var singleSegmentPool = sync.Pool{
-	New: func() any {
-		return singleSegment(nil)
-	},
+	bufferpool.Default.Put(*ssa)
+	(*ssa) = nil
```

Comment:
Nit: don't need the parens on the lvalue

Reply:
Ditto. Fixed in #465.